velox/functions/lib/tests/Re2FunctionsTest.cpp

```diff
@@ -384,6 +384,36 @@ TEST_F(Re2FunctionsTest, likePattern) {
   EXPECT_EQ(like("abc", "MEDIUM POLISHED%"), false);
 }
 
+TEST_F(Re2FunctionsTest, likePatternOptimizedCases) {
+  auto like = [&](std::optional<std::string> str,
+                  std::optional<std::string> pattern) {
+    return evaluateOnce<bool>("like(c0, '" + *pattern + "')", str);
+  };
+
+  EXPECT_EQ(like("abcde", "abcde"), true);
```

Comment:
Try logic with larger strings more than 16 characters in length also (these are stored differently). Look at other largeString tests for an idea.